### PR TITLE
Migration for the `sendTiming` events for `package:unified_analytics`

### PR DESCRIPTION
**Pull request by eliasyishak**
Related to tracker issue:
- https://github.com/flutter/flutter/issues/128251

image

The image above shows all of the instances where we have `sendTiming`. All of the call sites have been updated to use the new `Event.timing` event from `package:unified_analytics`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/runner/flutter_command_runner_test.dart [hunk 1]
```diff
@@ -144,14 +154,25 @@ void main() {
...
+        expect(testUsage.commands, contains(
+          const TestUsageCommand('version'),
+        ));
+        expect(fakeAnalytics.sentEvents, contains(
+          Event.flutterCommandResult(
+              commandPath: 'version',
+              result: 'success',
+              commandHasTerminal: false,
+            )
+        ));
```
Comment (eliasyishak) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_runner_test.dart:167]: Adding test cases for both legacy analytics and `package:unified_analytics`... we were previously not testing for the workflow when running `flutter --version`

**Comment by christopherfujino** [comment]
I believe this analysis error is unrelated to your change and fixed upstream:

```
Analyzing Á flutter sdk...                                      

  error • 'FakeView.render' ('void Function(Scene)') isn't a valid override of 'TestFlutterView.render' ('void Function(Scene, {Size? size})') • packages/flutter/test/widgets/multi_view_testing.dart:20:8 • invalid_override
  error • 'FakeView.render' ('void Function(Scene)') isn't a valid override of 'TestFlutterView.render' ('void Function(Scene, {Size? size})') • packages/flutter_test/test/multi_view_testing.dart:20:8 • invalid_override
   info • Place 'package:' imports before relative imports • packages/flutter_tools/lib/src/commands/packages.dart:22:1 • directives_ordering
```

**Comment by eliasyishak** [comment]
@christopherfujino ah i see, thanks, i was wondering why it kept coming up

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 2]
```diff
@@ -341,6 +342,11 @@ class FlutterCommandRunner extends CommandRunner<void> {
             status = version.toString();
           }
           globals.printStatus(status);
+          globals.analytics.send(Event.flutterCommandResult(
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart:345]: nit. can you put this line right after line 332, and also pass `FlutterGlobalOptions.kVersionFlag` as the commandPath to make it more obvious this is replicating line 332?

File: packages/flutter_tools/lib/src/web/compile.dart [hunk 3]
```diff
@@ -150,11 +150,17 @@ class WebBuilder {
       settings: buildSettingsString,
     ));
 
+    final Duration elapsedDuration = sw.elapsed;
     _flutterUsage.sendTiming(
       'build',
       compilerConfig.isWasm ? 'dart2wasm' : 'dart2js',
-      Duration(milliseconds: sw.elapsedMilliseconds),
+      elapsedDuration,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/compile.dart:157]: nice cleanup here

File: packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart [hunk 4]
```diff
@@ -209,12 +215,26 @@ void main() {
       const <String>['build', 'linux', '--no-pub']
     );
     expect(fileSystem.file('linux/flutter/ephemeral/generated_config.cmake'), exists);
+
+    expect(
+        analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'build',
+            variableName: 'cmake-linux'),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:224]:
```suggestion
    expect(
      analyticsTimingEventExists(
        sentEvents: fakeAnalytics.sentEvents,
        workflow: 'build',
        variableName: 'cmake-linux'),
      true,
    );
```

File: packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart [hunk 5]
```diff
@@ -209,12 +215,26 @@ void main() {
...
+    expect(
+        analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'build',
+            variableName: 'linux-ninja'),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_linux_test.dart:230]: nit

```suggestion
    expect(
      analyticsTimingEventExists(
        sentEvents: fakeAnalytics.sentEvents,
        workflow: 'build',
        variableName: 'linux-ninja',
      ),
      true,
    );
```

File: packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart [hunk 6]
```diff
@@ -194,11 +200,20 @@ STDERR STUFF
...
         const <String>['build', 'macos', '--no-pub']
     );
+
+    expect(
+        analyticsTimingEventExists(
+          sentEvents: fakeAnalytics.sentEvents,
+          workflow: 'build',
+          variableName: 'xcode-macos',
+        ),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart:210]: nit
```suggestion
      analyticsTimingEventExists(
        sentEvents: fakeAnalytics.sentEvents,
        workflow: 'build',
        variableName: 'xcode-macos',
      ),
      true,
    );
```

File: packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart [hunk 7]
```diff
@@ -209,6 +215,42 @@ void main() {
...
+    await createTestCommandRunner(command).run(
+      const <String>['windows', '--no-pub']
+    );
+
+    expect(
+        analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'build',
+            variableName: 'windows-cmake-generation'),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart:238]: nit

```suggestion
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'build',
          variableName: 'windows-cmake-generation',
        ),
        true,
      );
```

File: packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart [hunk 8]
```diff
@@ -209,6 +215,42 @@ void main() {
...
+    expect(
+        analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'build',
+            variableName: 'windows-cmake-build'),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart:244]:
```suggestion
      analyticsTimingEventExists(
        sentEvents: fakeAnalytics.sentEvents,
        workflow: 'build',
        variableName: 'windows-cmake-build',
      ),
      true,
    );
```

File: packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart [hunk 9]
```diff
@@ -145,11 +145,19 @@ void main() {
...
           Event.flutterBuildInfo(label: 'gradle-random-event-label-failure', buildType: 'gradle'),
         ]),
       );
+
+      expect(
+          analyticsTimingEventExists(
+              sentEvents: fakeAnalytics.sentEvents,
+              workflow: 'build',
+              variableName: 'gradle'),
+          true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart:159]:
```suggestion
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'build',
          variableName: 'gradle',
        ),
        true,
      );
```

File: packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart [hunk 10]
```diff
@@ -878,8 +886,16 @@ BuildVariant: paidProfile
         project: FlutterProject.fromDirectoryTest(fileSystem.currentDirectory),
       );
       expect(actual, <String>['freeDebug', 'paidDebug', 'freeRelease', 'paidRelease', 'freeProfile', 'paidProfile']);
+
+      expect(
+          analyticsTimingEventExists(
+              sentEvents: fakeAnalytics.sentEvents,
+              workflow: 'print',
+              variableName: 'android build variants'),
+          true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart:895]:
```suggestion
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'print',
          variableName: 'android build variants',
        ),
        true,
      );
```

File: packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart [hunk 11]
```diff
@@ -941,10 +957,18 @@ Gradle Crashed
         'freeDebug',
         project: FlutterProject.fromDirectoryTest(fileSystem.currentDirectory),
       );
+
+      expect(
+          analyticsTimingEventExists(
+              sentEvents: fakeAnalytics.sentEvents,
+              workflow: 'outputs',
+              variableName: 'app link settings'),
+          true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart:966]:
```suggestion
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'outputs',
          variableName: 'app link settings',
        ),
        true,
      );
```

File: packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart [hunk 12]
```diff
@@ -1010,8 +1034,16 @@ Gradle Crashed
         isFalse,
       );
       expect(processManager, hasNoRemainingExpectations);
+
+      expect(
+          analyticsTimingEventExists(
+              sentEvents: fakeAnalytics.sentEvents,
+              workflow: 'build',
+              variableName: 'gradle-aar'),
+          true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_gradle_builder_test.dart:1043]:
```suggestion
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'build',
          variableName: 'gradle-aar',
        ),
        true,
      );
```

File: packages/flutter_tools/test/general.shard/hot_test.dart [hunk 13]
```diff
@@ -335,6 +335,13 @@ void main() {
             scannedSourcesCount: 8
           )
         ));
+        expect(
+          analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'hot',
+            variableName: 'restart'),
+            true
+        );
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/hot_test.dart:344]:
```suggestion
        expect(
          analyticsTimingEventExists(
            sentEvents: fakeAnalytics.sentEvents,
            workflow: 'hot',
            variableName: 'restart',
          ),
          true,
        );
```

File: packages/flutter_tools/test/general.shard/hot_test.dart [hunk 14]
```diff
@@ -461,6 +468,13 @@ void main() {
             reloadVMTimeInMs: 512000
           ),
         ));
+        expect(
+          analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'hot',
+            variableName: 'reload'),
+            true
+        );
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/hot_test.dart:477]:
```suggestion
            variableName: 'reload',
          ),
          true,
        );
```

File: packages/flutter_tools/test/general.shard/ios/ios_device_start_nonprebuilt_test.dart [hunk 15]
```diff
@@ -135,6 +141,12 @@ void main() {
       expect(launchResult.started, false);
       expect(logger.errorText, contains('Xcode build is missing expected TARGET_BUILD_DIR build setting'));
       expect(processManager, hasNoRemainingExpectations);
+      expect(
+          analyticsTimingEventExists(
+              sentEvents: fakeAnalytics.sentEvents,
+              workflow: 'build',
+              variableName: 'xcode-ios'),
+          true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/ios/ios_device_start_nonprebuilt_test.dart:149]:
```suggestion
      expect(
        analyticsTimingEventExists(
          sentEvents: fakeAnalytics.sentEvents,
          workflow: 'build',
          variableName: 'xcode-ios',
        ),
        true,
      );
```

File: packages/flutter_tools/test/general.shard/resident_web_runner_test.dart [hunk 16]
```diff
@@ -693,6 +693,13 @@ void main() {
     expect(testUsage.timings, const <TestTimingEvent>[
       TestTimingEvent('hot', 'web-incremental-restart', Duration.zero),
     ]);
+    expect(fakeAnalytics.sentEvents, contains(
+      Event.timing(
+          workflow: 'hot',
+          variableName: 'web-incremental-restart',
+          elapsedMilliseconds: 0,
+        )
+    ));
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/resident_web_runner_test.dart:702]:
```suggestion
        workflow: 'hot',
        variableName: 'web-incremental-restart',
        elapsedMilliseconds: 0,
      ),
    ));
```

File: packages/flutter_tools/test/general.shard/resident_web_runner_test.dart [hunk 17]
```diff
@@ -779,6 +786,13 @@ void main() {
     expect(testUsage.timings, const <TestTimingEvent>[
       TestTimingEvent('hot', 'web-incremental-restart', Duration.zero),
     ]);
+    expect(fakeAnalytics.sentEvents, contains(
+      Event.timing(
+          workflow: 'hot',
+          variableName: 'web-incremental-restart',
+          elapsedMilliseconds: 0,
+        )
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/resident_web_runner_test.dart:794]:
```suggestion
        workflow: 'hot',
        variableName: 'web-incremental-restart',
        elapsedMilliseconds: 0,
      ),
```

File: packages/flutter_tools/test/general.shard/runner/flutter_command_runner_test.dart [hunk 18]
```diff
@@ -144,14 +154,25 @@ void main() {
...
         expect(version.didFetchTagsAndUpdate, true);
+        expect(testUsage.commands, contains(
+          const TestUsageCommand('version'),
+        ));
+        expect(fakeAnalytics.sentEvents, contains(
+          Event.flutterCommandResult(
+              commandPath: 'version',
+              result: 'success',
+              commandHasTerminal: false,
+            )
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_runner_test.dart:166]:
```suggestion
            commandPath: 'version',
            result: 'success',
            commandHasTerminal: false,
          ),
```

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart [hunk 19]
```diff
@@ -496,6 +505,16 @@ void main() {
...
+      // Iterate through and count all the [Event.timing] instances
+      int timingEventCounts = 0;
+      for (final Event e in fakeAnalytics.sentEvents) {
+        if (e.eventName == DashEvent.timing) {
+          timingEventCounts += 1;
+        }
+      }
+      expect(timingEventCounts, 0,
+          reason: 'There should not be any timing events sent, '
+              'there may be other events');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:517]:
```suggestion
      expect(
        timingEventCounts,
        0,
        reason: 'There should not be any timing events sent, there may be other events',
      );
```

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:517]: why would there be other events?

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:517]: oh, this test was only ever testing that there weren't timing events?

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart [hunk 20]
```diff
@@ -521,6 +540,14 @@ void main() {
           Duration(milliseconds: 500),
           label: 'success-blah1-blah2-blah3',
         )));
+      expect(fakeAnalytics.sentEvents, contains(
+        Event.timing(
+            workflow: 'flutter',
+            variableName: 'dummy',
+            elapsedMilliseconds: 500,
+            label: 'success-blah1-blah2-blah3',
+          )
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:549]:
```suggestion
          workflow: 'flutter',
          variableName: 'dummy',
          elapsedMilliseconds: 500,
          label: 'success-blah1-blah2-blah3',
        ),
```

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart [hunk 21]
```diff
@@ -545,6 +572,14 @@ void main() {
           label: 'fail',
         ),
       ));
+      expect(fakeAnalytics.sentEvents, contains(
+        Event.timing(
+            workflow: 'flutter',
+            variableName: 'dummy',
+            elapsedMilliseconds: 1000,
+            label: 'fail',
+          )
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:581]:
```suggestion
          workflow: 'flutter',
          variableName: 'dummy',
          elapsedMilliseconds: 1000,
          label: 'fail',
        ),
```

File: packages/flutter_tools/test/general.shard/web/compile_web_test.dart [hunk 22]
```diff
@@ -124,6 +124,12 @@ void main() {
     final TestTimingEvent timingEvent = testUsage.timings.single;
     expect(timingEvent.category, 'build');
     expect(timingEvent.variableName, 'dart2wasm');
+    expect(
+        analyticsTimingEventExists(
+            sentEvents: fakeAnalytics.sentEvents,
+            workflow: 'build',
+            variableName: 'dart2wasm'),
+        true);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/web/compile_web_test.dart:132]:
```suggestion
      analyticsTimingEventExists(
        sentEvents: fakeAnalytics.sentEvents,
        workflow: 'build',
        variableName: 'dart2wasm',
      ),
      true,
    );
```

File: packages/flutter_tools/test/src/common.dart [hunk 23]
```diff
@@ -347,3 +348,36 @@ FakeAnalytics getInitializedFakeAnalyticsInstance({
     clientIde: clientIde,
   );
 }
+
+/// This utility function allows us to check for an instance of
```
Comment (christopherfujino) [inline packages/flutter_tools/test/src/common.dart:352]: Please ensure all dartdocs start with a single sentence separated from the rest of the text:

https://dart.dev/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

https://dart.dev/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

**Review by christopherfujino — APPROVED** [review]
This LGTM modulo nits

File: packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart (hunk 19 above)
Comment (eliasyishak) [reply packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart:517]: Correct, with package:usage, we had separate lists for timing, commands, and other events, but with the new approach they are stored in the same `sentEvents` list.

**Comment by eliasyishak** [comment]
This PR was reverted due to Google3 updates that need to be made:
- #139278

**Pull request merged**